lib/pxproxy.js

```diff
@@ -26,35 +37,52 @@ function getClient(req, pxConfig, ip, cb) {
         callData.headers[pxConfig.FIRST_PARTY_HEADER] = 1;
         callData.keepAsBuffer = true;
         request.get(callData, (error, response, body) => {
-            response = {
-                status: 200,
-                headers: response.headers,
-                body: body
-            };
-            return cb(response);
+            try {
+                res = {
+                    status: 200,
+                    headers: response.headers,
```

Comment:
instead of try-catch just check if the response is empty and return an error if needed